steps/pipeline-tests-jest.yaml

```diff
@@ -0,0 +1,43 @@
+parameters:
+- name: environment
+  default: ''
+- name: domain_prefix
+  default: ''
+- name: test_file
+  default: ''
+- name: workingDirectory
+  default: ''
+- name: junit_output_dir
+  default: ''
+- name: test_title
+  default: ''
+- name: dns_zone
+  default: platform.hmcts.net
+- name: app_name
+  default: ''
+
+steps:
+- task: Bash@3
+  displayName: Run pipeline tests
+  inputs:
+    targetType: 'inline'
+    script: |
+      mkdir -p ${{ parameters.junit_output_dir }}
+      yarn
+      yarn test-junit
+    workingDirectory: '${{ parameters.workingDirectory }}'
+    failOnStderr: false
+  env:
+    JEST_JUNIT_OUTPUT_DIR: ${{ parameters.junit_output_dir }}
+    JEST_JUNIT_OUTPUT_NAME: report_$(system.jobId).xml
+    APP_NAME: ${{ parameters.app_name }}
+    TEST_URL: '${{ parameters.app_name }}.${{ parameters.dns_zone }}'
```

Comment:
this is missing the environment and protocol

and needs to remove it on prod

Reply:
The protocol is in the test itself.

The environment isn't being used since they mostly don't match the dns zone names e.g. test is perftest.

At the moment, I'm setting these in the consuming repo's pipeline config:

app_name: https://github.com/hmcts/azure-platform-terraform/pull/1231/files#diff-a74aef3016a9caa7857d8636a196d2ca33561aeca7ddee0b93f7fedf595151abR45

dns_zone: https://github.com/hmcts/azure-platform-terraform/pull/1231/files#diff-a74aef3016a9caa7857d8636a196d2ca33561aeca7ddee0b93f7fedf595151abR66

Reply:
hmm annoying, the tests shouldn't need to know this.

do you need any of this info then or can the tests just handle it for now?

Reply:
I think the tests are grand for now. The required parameters will be passed through by the consuming pipeline and the tests are not enabled on prod - would fail anyway if the required parameters were added since plum isn't on prod.

Reply:
i mean do you need these app_name, test_url, dns_zone parameters?

Reply:
yeah, they enable the pipeline output and test_url to be customised by the consuming pipeline

Reply:
i think this test_url as is should be removed as it's not a real url, i.e. doesn't include the protocol and the environment.

if you need the dns_zone then can keep that